crates/sui-indexer/migrations_v2/2023-08-19-044035_tx_indices/up.sql

```diff
@@ -0,0 +1,33 @@
+CREATE TABLE tx_indices (
+    tx_sequence_number          BIGINT       PRIMARY KEY,
+    checkpoint_sequence_number  BIGINT       NOT NULL,
+    -- bytes of the transaction digest
+    transaction_digest          bytea        NOT NULL,
+    -- array of ObjectID in bytes.
+    input_objects               bytea[]      NOT NULL,
+    -- array of ObjectID in bytes
+    changed_objects             bytea[]      NOT NULL,
+    -- array of SuiAddress in bytes. All signers of the transaction.
+    senders                     bytea[]      NOT NULL,
+    -- array of SuiAddress in bytes. All gas owners of the transaction.
+    payers                      bytea[]      NOT NULL,
+    -- array of SuiAddress in bytes. 
+    recipients                  bytea[]      NOT NULL,
+    -- array of PackageID in bytes of all MoveCalls of the transaction.
+    packages                    bytea[]      NOT NULL,
+    -- array of "package::module" of all MoveCalls of the transaction.
+    -- e.g. "0x0000000000000000000000000000000000000000000000000000000000000003::sui_system"
+    package_modules             text[]       NOT NULL,
+    -- array of "package::module::function" of all MoveCalls of the transaction.
+    -- e.g. "0x0000000000000000000000000000000000000000000000000000000000000003::sui_system::request_add_stake"
+    package_module_functions    text[]       NOT NULL
+);
+
+CREATE INDEX tx_indices_input_objects ON tx_indices USING GIN(input_objects);
+CREATE INDEX tx_indices_changed_objects ON tx_indices USING GIN(changed_objects);
+CREATE INDEX tx_indices_senders ON tx_indices USING GIN(senders);
+CREATE INDEX tx_indices_recipients ON tx_indices USING GIN(recipients);
```

Comment:
I think we talked about not including an index on the other kinds of address to transaction relationships, but in hindsight, it seems like it would be pretty impossible to use them in a query without such an index -- should we add one for payers?

Reply:
if that's a required feature then we have to create an index for it. Just to caveat that each index will add more burden to the ingestion pipeline, so we'd better be self-disciplined in adding them

Reply:
Would it be feasible to add it and then gate the code that handles post-processing so if a particular deployment doesn't want to deal with indexing address-transaction relationships, it can turn that bit off? We will have a similar functionality on the GraphQL side to disable the relevant aspects of the schema.

Reply:
We can check existence of table indices and create if needed when the instance starts. So it's doable, just not by diesel.